pkg/shared/util/wait_until_ready.go

```diff
@@ -0,0 +1,7 @@
+package util
```

Comment:
Does it make sense to put it in `pkg/metrics` with interface name `HealthChecker`? Or put it in `pkg/shared/interfaces`?

Reply:
The WaitUntilReady is not only used in the metrics server/metrics pkg. It's also used when we start the grpc connection. 
I renamed it to `ReadyChecker`.. 
`pkg/shared/interfaces` or `pkg/shared/util`? I prefer `util` though...